langchain4j-google-ai-gemini/src/main/java/dev/langchain4j/model/googleai/GoogleAiTokenizer.java

```diff
@@ -0,0 +1,135 @@
+        this.apiKey = ensureNotBlank(apiKey, "apiKey");
+        this.maxRetries = getOrDefault(maxRetries, 3);
+        this.geminiService = GeminiService.getGeminiService(
+            logRequestsAndResponses ? log : null,
+            duration != null ? duration : Duration.ofSeconds(60));
+    }
+
+    @Override
+    public int estimateTokenCountInText(String text) {
+        return estimateTokenCountInMessages(singletonList(UserMessage.from(text)));
+    }
+
+    @Override
+    public int estimateTokenCountInMessage(ChatMessage message) {
+        return estimateTokenCountInMessages(singletonList(message));
+    }
+
+    @Override
+    public int estimateTokenCountInToolExecutionRequests(Iterable<ToolExecutionRequest> toolExecutionRequests) {
+        List<ToolExecutionRequest> allToolRequests = new LinkedList<>();
+        toolExecutionRequests.forEach(allToolRequests::add);
+
+        return estimateTokenCountInMessage(AiMessage.from(allToolRequests));
+    }
+
+    @Override
+    public int estimateTokenCountInMessages(Iterable<ChatMessage> messages) {
+        List<ChatMessage> allMessages = new LinkedList<>();
+        messages.forEach(allMessages::add);
+
+        List<GeminiContent> geminiContentList = fromMessageToGContent(allMessages, null);
+
+        GeminiCountTokensRequest countTokensRequest = new GeminiCountTokensRequest();
+        countTokensRequest.setContents(geminiContentList);
+
+        return estimateTokenCount(countTokensRequest);
+    }
+
+    @Override
+    public int estimateTokenCountInToolSpecifications(Iterable<ToolSpecification> toolSpecifications) {
+        List<ToolSpecification> allTools = new LinkedList<>();
+        toolSpecifications.forEach(allTools::add);
+
+        GeminiContent dummyContent = GeminiContent.builder().parts(
+            singletonList(GeminiPart.builder()
+                .text("Dummy content")
+                .build())
+        ).build();
+
+        GeminiCountTokensRequest countTokensRequestWithDummyContent = new GeminiCountTokensRequest();
+        countTokensRequestWithDummyContent.setGenerateContentRequest(GeminiGenerateContentRequest.builder()
+            .model("models/" + this.modelName)
+            .contents(singletonList(dummyContent))
+            .tools(FunctionMapper.fromToolSepcsToGTool(allTools, false))
+            .build());
+
+        GeminiCountTokensRequest countTokensRequestForDummyContent = new GeminiCountTokensRequest();
+        countTokensRequestForDummyContent.setGenerateContentRequest(GeminiGenerateContentRequest.builder()
+            .model("models/" + this.modelName)
+            .contents(singletonList(dummyContent))
+            .build());
+
+        // The API doesn't allow us to make a request to count the tokens of the tool specifications only.
+        // Instead, we take the approach of adding a dummy content in the request, and subtract the tokens for the dummy request.
+        // This is generating 2 API calls.
```

Comment:
I assume dummy content token count will always be the same (2 tokens in this case), so perharps we could juse subtract 2 and avoid second call?